src/coreclr/src/vm/interpreter.cpp

```diff
@@ -1636,6 +1636,7 @@ void Interpreter::JitMethodIfAppropriate(InterpreterMethodInfo* interpMethInfo,
 
         if (InterpretationStubToMethodInfo(stub) == md)
         {
+            CodeVersionManager::LockHolder _lockHolder;
```

Comment:
Its best to do as little work inside of locks as necessary. I'd recommend moving it down to after the ifdef at 1667.

Reply:
Done in commit https://github.com/dotnet/runtime/pull/37825/commits/748c7ade014aeb6dbe5246d437ac1eb95162b58b.